### ISSUE TITLE
False positive: empty_line_after_outer_attribute incorrectly flags deny attribute

### ISSUE THREAD

**Issue by alycda** [rust-lang/rust-clippy#15525]
**Summary**

The `empty_line_after_outer_attribute` lint incorrectly flags a deny attribute as having an empty line after it, when the actual outer attribute (`#[instrument]`) does not have an empty line.

**Lint Name**
`empty_line_after_outer_attribute`

**Reproducer**
```rust
//! Minimal reproduction case for clippy false positive

use tracing::instrument;

// This line triggers false positive "empty line after outer attribute"  
#[deny(clippy::print_stdout, clippy::print_stderr)]

/// Example function
#[instrument]
pub extern "C" fn example_function() -> i32 {
    42
}
```

**Expected vs Actual Behavior**
- **Expected**: No warning, as `#[instrument]` (line 9) has no empty line after it
- **Actual**: Warning at line 6: "empty line after outer attribute"

**Version**
- rustc 1.89.0 (29483883e 2025-08-04)
- clippy 0.1.89 (29483883ee 2025-08-04)

**Additional Context**
The lint appears to be incorrectly analyzing the `#[deny(...)]` attribute instead of the actual outer attribute `#[instrument]` that precedes the function declaration.

**Comment by alycda** [comment]
I see the inner/outer attribute is incorrect, `#!` must come BEFORE `//!`

```
#![deny(clippy::print_stdout, clippy::print_stderr)]
//! Minimal reproduction case for clippy false positive

use tracing::instrument;

/// Example function
#[instrument]
pub extern "C" fn example_function() -> i32 {
    42
}
```
